packages/components/notification/src/NotificationProvider.tsx

```diff
@@ -0,0 +1,322 @@
+//     }
+//   })
+//
+//   return placementNotifications
+// }
+
+function usePlacementNotifications(
+  props: NotificationProviderProps,
+  config: NotificationConfig,
+  notifications: Ref<NotificationOptions[]>,
+) {
+  return computed(() => {
+    const initValue = getPlacementInitValue<NotificationOptions[]>(() => []) as NotificationPlacementMap
+    return notifications.value.reduce((acc, item) => {
+      const notificationProps = {
+        ...getRealObj(omit<NotificationConfig, 'maxCount'>(config, ['maxCount'])),
+        ...getRealObj(omit<NotificationProviderProps, 'maxCount'>(props, ['maxCount'])),
+        ...getRealObj(omit<NotificationOptions, 'onDestroy'>(item, ['onDestroy'])),
+      } as NotificationProps
+      const placement = notificationProps.placement!
+      acc[placement].push(notificationProps)
+      return acc
+    }, initValue)
+  })
+}
+
+function useNotification(maxCount: ComputedRef<number>) {
+  const notifications = ref<NotificationOptions[]>([])
+
+  const getCurrIndex = (key: VKey) => {
+    return notifications.value.findIndex(notification => notification.key === key)
+  }
+
+  const add = (item: NotificationOptions) => {
+    const currIndex = item.key ? getCurrIndex(item.key) : -1
+    if (currIndex !== -1) {
+      notifications.value.splice(currIndex, 1, item)
+      return item.key!
+    }
+
+    if (notifications.value.length >= maxCount.value) {
+      notifications.value = notifications.value.slice(-maxCount.value + 1)
+    }
+
+    const key = item.key ?? uniqueId('ix-notification')
+    notifications.value.push({ ...item, key })
+    return key
+  }
+
+  const update = (key: VKey, item: Partial<NotificationOptions>) => {
+    const currIndex = getCurrIndex(key)
+    if (currIndex !== -1) {
+      const newItem = { ...notifications.value[currIndex], ...item }
+      notifications.value.splice(currIndex, 1, newItem)
+    }
+  }
+
+  const destroy = (key: VKey | VKey[]) => {
+    const keys = convertArray(key)
+    keys.forEach(key => {
+      const currIndex = getCurrIndex(key)
+      if (currIndex !== -1) {
+        const item = notifications.value.splice(currIndex, 1)
+        callEmit(item[0].onDestroy, key)
+      }
+    })
+  }
+
+  const destroyAll = () => {
+    notifications.value = []
+  }
+
+  const loadContainer = ref(false)
+
+  const open = (options: NotificationOptions): NotificationRef => {
+    const key = add(options)
+    const ref = {
+      key,
+      update: (options: Partial<NotificationOptions>) => update(key, options),
+      destroy: () => destroy(key),
+    }
+    if (!loadContainer.value) {
+      loadContainer.value = true
+    }
+    return ref
+  }
+
+  const { info, success, warning, error } = notificationType.reduce(
+    (acc, type) => ({
+      ...acc,
+      [type]: (options: Omit<NotificationOptions, 'type'>) => open({ ...options, type }),
+    }),
+    {} as Record<NotificationType, (options: Omit<NotificationOptions, 'type'>) => NotificationRef>,
+  )
```

Comment:
```suggestion
  const  notificationTypes = ['info', 'success', 'warning', 'error'] as const
  const [info, success, warning, error] = notificationTypes.map(type => {
    return (options?: Omit<NotificationOptions, 'type'>) => open({ ...options, type })
  })
```
要沉迷 reduce 呢，map 不香吗？

Reply:
fix